decidim-debates/app/commands/decidim/debates/update_debate.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Debates
+    # A command with all the business logic when a user updates a debate.
+    class UpdateDebate < Rectify::Command
+      # Public: Initializes the command.
+      #
+      # form         - A form object with the params.
+      # current_user - The current user.
+      # debate - the debate to update.
+      def initialize(form, current_user, debate)
+        @form = form
+        @current_user = current_user
+        @debate = debate
+      end
+
+      # Executes the command. Broadcasts these events:
+      #
+      # - :ok when everything is valid, together with the debate.
+      # - :invalid if the form wasn't valid and we couldn't proceed.
+      #
+      # Returns nothing.
+      def call
+        return broadcast(:invalid) if form.invalid?
+        return broadcast(:invalid) unless debate.editable_by?(current_user)
+
+        update_debate
+        broadcast(:ok, debate)
+      end
+
+      private
+
+      attr_reader :form, :debate, :current_user
+
+      def organization
+        @organization ||= form.current_component.organization
+      end
+
+      def i18n_field(field)
```

Comment:
Instead of doing this, please set only the `I18n.locale` version:

```ruby
{ I18n.locale => field }
```

The translation helpers can already handle this case.

Otherwise we won't be able to translate this in #6127 :(

Reply:
This should be fixed for the `CreateDebate` command too!

Reply:
We'll write a migration to fix the already existing data 😄 (we'll do that in another PR)

Reply:
Done!